erpnext/assets/doctype/asset/asset.js

```diff
@@ -31,6 +31,9 @@ frappe.ui.form.on('Asset', {
 		frm.events.make_schedules_editable(frm);
 
 		if (frm.doc.docstatus==1) {
+			if (frm.doc.maintenance_required) {
+				frm.trigger("update_maintenance_status");
```

Comment:
Don't call this type of functions in `refresh`. Instead you should write a function to update the asset status and call it via Scheduler using hooks.

Reply:
@nabinhait Done via hooks. Please check if ok.